azure group controller crashes when mail is nil

```
[mjudeiki@redhat openshift-azure]$ oc logs -f customer-admin-controller-776684f766-8gs4w
time="2019-04-12T12:35:42Z" level=info msg="azure-controller pod starting, git commit v4.0-154-gbf3371a-clean"
time="2019-04-12T12:35:42Z" level=info msg="starting manager"
time="2019-04-12T12:35:43Z" level=debug msg="NSC: reconciling namespace e2e-test-qmtf8"
time="2019-04-12T12:35:43Z" level=debug msg="NSC: creating rolebinding e2e-test-qmtf8/osa-customer-admin"
time="2019-04-12T12:35:43Z" level=debug msg="RBC: reconciling rolebinding e2e-test-qmtf8/osa-customer-admin"
time="2019-04-12T12:35:43Z" level=debug msg="RBC: reconciling rolebinding e2e-test-qmtf8/osa-customer-admin-project"
time="2019-04-12T12:35:43Z" level=debug msg="NSC: creating rolebinding e2e-test-qmtf8/osa-customer-admin-project"
time="2019-04-12T12:36:42Z" level=debug msg="AAD Group Reconciler (poll).."
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x105a8e9]
goroutine 76 [running]:
github.com/openshift/openshift-azure/pkg/controllers/customeradmin.fromMSGraphGroup(0xc0000adc80, 0x0, 0x130fe98, 0x13, 0xc000254c60, 0x2, 0x2, 0x0, 0x0)                                                                                                                                
        /go/src/github.com/openshift/openshift-azure/pkg/controllers/customeradmin/group_mapping.go:34 +0x129
github.com/openshift/openshift-azure/pkg/controllers/customeradmin.updateKubeGroup(0xc0000adc80, 0x14847c0, 0xc0003c0440, 0x130fe98, 0x13, 0xc000254c60, 0x2, 0x2, 0x0, 0x1)                                                                                                             
        /go/src/github.com/openshift/openshift-azure/pkg/controllers/customeradmin/group_reconcile.go:23 +0x126
github.com/openshift/openshift-azure/pkg/controllers/customeradmin.reconcileGroups(0xc0000adc80, 0x14644a0, 0xc00033ea00, 0x14847c0, 0xc0003c0440, 0xc0004367b0, 0x42e4fb, 0x138adc8)                                                                                                    
        /go/src/github.com/openshift/openshift-azure/pkg/controllers/customeradmin/group_reconcile.go:53 +0xfa
github.com/openshift/openshift-azure/pkg/controllers/customeradmin.(*reconcileGroup).pollEvent(0xc000427080, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/openshift/openshift-azure/pkg/controllers/customeradmin/group_controller.go:100 +0xad
github.com/openshift/openshift-azure/pkg/controllers/customeradmin.(*reconcileGroup).pollEvent-fm(0x0, 0x0, 0x0, 0x0, 0xc00037f288)
        /go/src/github.com/openshift/openshift-azure/pkg/controllers/customeradmin/group_controller.go:75 +0x52
github.com/openshift/openshift-azure/vendor/sigs.k8s.io/controller-runtime/pkg/predicate.Funcs.Generic(0x0, 0x0, 0x0, 0xc0003c0580, 0x0, 0x0, 0x0, 0x0, 0x101)                                                                                                                           
        /go/src/github.com/openshift/openshift-azure/vendor/sigs.k8s.io/controller-runtime/pkg/predicate/predicate.go:86 +0x53
github.com/openshift/openshift-azure/vendor/sigs.k8s.io/controller-runtime/pkg/source.(*Channel).Start.func2(0xc0004444e0, 0xc0003c0590, 0x1, 0x1, 0x14809c0, 0x1fc0c18, 0x148f260, 0xc00042cf20)                                                                                        
        /go/src/github.com/openshift/openshift-azure/vendor/sigs.k8s.io/controller-runtime/pkg/source/source.go:168 +0x18e
created by github.com/openshift/openshift-azure/vendor/sigs.k8s.io/controller-runtime/pkg/source.(*Channel).Start
        /go/src/github.com/openshift/openshift-azure/vendor/sigs.k8s.io/controller-runtime/pkg/source/source.go:164 +0x120
```

--- a/pkg/controllers/customeradmin/group_mapping.go
+++ b/pkg/controllers/customeradmin/group_mapping.go
@@ -31,7 +31,11 @@ func fromMSGraphGroup(log *logrus.Entry, kubeGroup *v1.Group, kubeGroupName stri
 	}
 	g.Users = []string{}
 	for _, user := range msGroupMembers {
-		g.Users = append(g.Users, *user.Mail)
+		if user.UserType == graphrbac.Guest && user.Mail != nil {
+			g.Users = append(g.Users, *user.Mail) // doesn't include #EXT#
+		} else {
+			g.Users = append(g.Users, *user.UserPrincipalName)
+		}
 	}
 	sort.Strings(g.Users)
 	return g, !reflect.DeepEqual(kubeGroup, g)

--- a/pkg/controllers/customeradmin/group_mapping_test.go
+++ b/pkg/controllers/customeradmin/group_mapping_test.go
@@ -38,7 +38,7 @@ func TestFromMSGraphGroup(t *testing.T) {
 			kubeGroupName: osaCustomerAdmins,
 			msGroupMembers: []graphrbac.User{
 				{
-					Mail: to.StringPtr("foo@somewhere.com"),
+					UserPrincipalName: to.StringPtr("foo@somewhere.com"),
 				},
 			},
 			want: &v1.Group{
@@ -59,7 +59,7 @@ func TestFromMSGraphGroup(t *testing.T) {
 			},
 			msGroupMembers: []graphrbac.User{
 				{
-					Mail: to.StringPtr("foo@somewhere.com"),
+					UserPrincipalName: to.StringPtr("foo@somewhere.com"),
 				},
 			},
 			want: &v1.Group{
@@ -81,10 +81,10 @@ func TestFromMSGraphGroup(t *testing.T) {
 			},
 			msGroupMembers: []graphrbac.User{
 				{
-					Mail: to.StringPtr("foo@somewhere.com"),
+					UserPrincipalName: to.StringPtr("foo@somewhere.com"),
 				},
 				{
-					Mail: to.StringPtr("tim@somewhere.com"),
+					UserPrincipalName: to.StringPtr("tim@somewhere.com"),
 				},
 			},
 			want: &v1.Group{
@@ -106,7 +106,8 @@ func TestFromMSGraphGroup(t *testing.T) {
 			},
 			msGroupMembers: []graphrbac.User{
 				{
-					Mail: to.StringPtr("foo@somewhere.com"),
+					Mail:     to.StringPtr("foo@somewhere.com"),
+					UserType: graphrbac.Guest,
 				},
 			},
 			want: &v1.Group{

